### DISCUSSION TITLE
nullable BelongsToSelect

### DISCUSSION THREAD

**Discussion by loranger** [filamentphp/filament#747] [Help]
Hello,

I have a resource which _may have_ a single relation.
I added a `Components\BelongsToSelect::make('parent_id')->relationship('parent', 'name')` component, but I now have to add a relation in order to save my resource otherwise I get a validation error.

Do you know how I can make this relation optional ?

**Comment by danharrin** [comment]
It should be optional, so that's a bug. Is this a v1 admin resource or v2?

**Reply by loranger** [reply]
It's a v1 admin (I can't wait to try the v2, by the way ;) )
Can I bypass this validation, maybe from the EditCategory file ?

**Reply by danharrin** [reply]
Is there a `nullable()` method in v1? I can't remember haha.

**Reply by loranger** [reply]
[It seems](nullable) you had added it, yes ;)
But it does not affect my form and the entry is not validated

**Reply by danharrin** [reply]
In the `beforeValidate()` hook of the page, can you `dd($this->rules())` and see if there's a `required` rule on that field or something?

**Reply by loranger** [reply]
I had an error using `$this->rules()`, but `$this->getRules()` return all this:

```
array:8 [▼
  "record.name" => array:1 [▶]
  "record.sku" => array:1 [▶]
  "record.parent_id" => array:2 [▼
    0 => "exists:App\Models\Category,id"
    1 => "nullable"
  ]
  "record.description" => array:1 [▶]
  "record.cover_path" => array:1 [▶]
  "temporaryUploadedFiles.record.cover_path" => array:3 [▶]
  "record.hidden" => array:1 [▶]
  "record.active" => array:1 [▶]
]

```

**Reply by danharrin** [reply]
Interesting, so are `exists` and `nullable` not compatible in Laravel?

**Reply by loranger** [reply]
I also tried with `sometimes`, but it fails as soon as an `exists` rule is added:

```php
->removeRules('exists')
->rules(['sometimes', 'nullable', 'exists:App\Models\Category,id']),
```

**Reply by danharrin** [reply]
Apologies for this, absolutely no clue why Laravel is doing this 🤷

**Reply by loranger** [reply]
No problem, `removeRules` helped me. If I have time, I'll digg a little bit deeper in laravel rules in order to understand

**Comment by loranger** [comment]
Hi,

Since I migrated to Filament v2
I cannot use `removeRules` anymore.

Is there a workaround, @danharrin ?

**Reply by danharrin** [reply]
Not really at the moment, did you investigate why this is not working in Laravel? Can you replicate it in a Laravel playground or anything?
